bin/oneoff/migrate_db.rb

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env ruby
+
+# One-off script to migrate live application to a new db hostname.
+
+require 'json'
+require 'uri'
+
+# Update database endpoint Chef Attributes in the environment.
+def update_endpoints(new_hostname, environment)
+  puts "Updating #{environment} DB endpoints to #{new_hostname}..."
+  environment_json = JSON.parse(`knife environment show #{environment} --format json`)
+  reader, writer = %w(reader writer).map do |db|
+      URI.parse(environment_json['override_attributes']['cdo-secrets']["db_#{db}"]).tap do |uri|
+        uri.hostname = new_hostname
+        environment_json['override_attributes']['cdo-secrets']["db_#{db}"] = uri.to_s
+      end
+  end
+  Tempfile.open(['environment','.json']) do |tmp|
+    File.write(tmp.path, environment_json.to_json)
+    `knife environment from file #{tmp.path} -E #{environment}`
+  end
+  [reader, writer]
+end
+
+# Executes `cmd` in a subprocess, streaming the output to stdout.
+def run(cmd)
+  IO.popen(cmd) { |output| output.each { |line| puts line.chomp } }
+end
+
+# Use the `pdsh` utility and `ls_frontend_servers` script to 'fan out' a command via ssh to all front-ends.
+def fanout(cmd)
+  puts "Running #{cmd}..."
+  servers = JSON.parse(`../bin/ls_frontend_servers -j`).values
+  run "pdsh -w #{servers.join(',')} -R ssh #{cmd}"
+end
+
+def main
+  db_host=ARGV[0]
+  environment=ARGV[1]
+
+  # 1. Stop applications on all frontends. [live-site downtime begins]
+  fanout 'sudo service dashboard stop; sudo service pegasus stop'
```

Comment:
If this script is to live in our repository, I'd strongly encourage a 'Do you mean to do this?' confirmation before causing live-site downtime.